extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -708,20 +713,40 @@ export class AmpConsent extends AMP.BaseElement {
 
   /**
    * If consentString is undefined or invalid, don't
-   * include any metadata in update.
+   * include any metadata in update. Otherwise, convert to
+   * to ConsentMetadataDef
    * @param {JsonObject=} opt_metadata
    * @param {string=} opt_consentString
-   * @return {?JsonObject|undefined}
+   * @return {ConsentMetadataDef|undefined}
    */
   configureMetadataByConsentString_(opt_metadata, opt_consentString) {
+    if (!opt_metadata) {
+      return;
+    }
     if (!isObject(opt_metadata) || !opt_consentString) {
       user().error(
         TAG,
         'CMP metadata is invalid or no consent string is found.'
       );
       return;
     }
-    return opt_metadata;
+    this.assertMetadataValues(opt_metadata);
+    return constructMetadata(opt_metadata['consentStringType']);
+  }
+
+  /**
+   * Confirm that the metadata values are valid. Remove otherwise.
+   * @param {JsonObject} metadata
+   */
+  assertMetadataValues(metadata) {
```

Comment:
We should consider moving the method to `consent-config.js`. 
It's not blocking, feel free to do so in a separate PR.

Reply:
Yes that sounds like a good idea.